### PR TITLE
Cleanup MonthCalendar interop definitions

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Make `MCGRIDINFO` blittable by using `Span`
- Cleanup `MCHITTESTINFO` interop

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2630)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs [hunk 1]
```diff
@@ -461,85 +460,89 @@ private CalendarRowAccessibleObject GetCalendarRow(int calendarIndex, Accessible
...
+
+                    bool result = GetCalendarGridInfoText(ref gridInfo);
+                    text = name.SliceAtFirstNull().ToString();
+                    return result;
+                }
             }
 
             // Use to retrieve MCGIF_NAME only.
-            private bool GetCalendarGridInfoText(ref ComCtl32.MCGRIDINFO gridInfo)
+            private bool GetCalendarGridInfoText(ref MCGRIDINFO gridInfo)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs]: This method is only called from one place, so it is worth inlining it there.

File: src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs [hunk 2]
```diff
@@ -461,85 +460,89 @@ private CalendarRowAccessibleObject GetCalendarRow(int calendarIndex, Accessible
...
+                        iCol = column,
+                        iRow = row,
+                        pszName = pName,
+                        cchName = (uint)name.Length - 1
+                    };
+
+                    bool result = GetCalendarGridInfoText(ref gridInfo);
+                    text = name.SliceAtFirstNull().ToString();
+                    return result;
+                }
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs]: Could you please a test(s) to verify the behaviour remains in tact.
Looks like we can test it via `GetCalendarChildName` method.

Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/MonthCalendar.MonthCalendarAccessibleObject.cs]: TODO

**Comment by hughbe** [comment]
@RussKie I think all your feedback has been addressed FYI

File: src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/MonthCalendar.CalendarHeaderAccessibleObjectTests.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+{
+    public class MonthCalendarCalendarHeaderAccessibleObjectTests : IClassFixture<ThreadExceptionFixture>
+    {
+        [WinFormsFact]
+        public void CalendarHeaderAccessibleObjectTests_Name_Get_ReturnsExpected()
+        {
+            using var calendar = new MonthCalendar();
+            MonthCalendar.MonthCalendarAccessibleObject accessibleObject = Assert.IsType<MonthCalendar.MonthCalendarAccessibleObject>(calendar.AccessibilityObject);
+            var headerAccessibleObject = new MonthCalendar.CalendarHeaderAccessibleObject(accessibleObject, 0);
+            Assert.Empty(headerAccessibleObject.Name);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/AccessibleObjects/MonthCalendar.CalendarHeaderAccessibleObjectTests.cs]: This test doesn't look like it thoroughly exercises the change in `GetCalendarGridInfoText` if the string is empty.

**Comment by RussKie** [comment]
@M-Lipin please review. Could you also suggest how to test `GetCalendarGridInfoText` method properly?

**Comment by M-Lipin** [comment]
Reviewed the changes. Look good to me in general. Was not able to test these with the app due to missing System.Windows.Forms.Primitives.dll in the latest SDK build. But seems that the updates should be OK. @hughbe please add the test for `GetCalendarGridInfoText` to be sure that the garbage collection works as expected with introduced refactoring.

**Comment by hughbe** [comment]
> But seems that the updates should be OK. @hughbe please add the test for GetCalendarGridInfoText to be sure that the garbage collection works as expected with introduced refactoring.

Thats what the test I added does:
```cs
public override string Name => _calendarAccessibleObject.GetCalendarChildName(_calendarIndex, 
CalendarChildType.CalendarHeader);
```

```cs
public string GetCalendarChildName(int calendarIndex, CalendarChildType calendarChildType, AccessibleObject parentAccessibleObject = null, int index = -1)
{
    switch (calendarChildType)
    {
        case CalendarChildType.CalendarHeader:
            GetCalendarGridInfoText(ComCtl32.MCGIP.CALENDARHEADER, calendarIndex, 0, 0, out string text);
            return text;
        case CalendarChildType.TodayLink:
            return string.Format(SR.MonthCalendarTodayButtonAccessibleName, _owner.TodayDate.ToShortDateString());
    };

    return string.Empty;
}
```

**Comment by RussKie** [comment]
What happened here? There's only a new test

**Comment by hughbe** [comment]
Should be working now. Forgot to push!

**Pull request merged**